lib/browser/api/menu.js

```diff
@@ -176,6 +172,16 @@ Menu.buildFromTemplate = function (template) {
 
 /* Helper Functions */
 
+// validate the template against having the wrong attribute
+function typeCheckTemplate (template) {
+  template.forEach((item) => {
+    if ((item === null) || (typeof item !== 'object') ||
+        (!item.label && !item.role && !item.type)) {
+      throw new TypeError('Invalid template for MenuItem')
```

Comment:
would be great to add a little more info to this message, e.g.

`Invalid template for MenuItem: must have at least one of label, role or type`